### DISCUSSION TITLE
Is there a way to create a new order through API without an API key?

### DISCUSSION THREAD

**Discussion by forkata** [solidusio/solidus#4959] [Troubleshooting] [answered]
This [recent issue](https://github.com/solidusio/solidus/pull/4958) reminded me of a case I encountered recently where we needed to generate a new guest order and add items to it. Since we don't have a logged-in user in that case, we don't have an API key with which we can call the [create order API](https://solidus.stoplight.io/docs/solidus/c7dfbf0398c40-create-order).

This is possible through the `/orders/populate` controller that `spree_frontend` provides and probably there is a way to do it with `solidus_starter_frontend`, but I am curious if there is a solution for doing this purely through the API?

**Comment by kennyadsl** [comment, marked as answer]
@forkata I think `POST /orders` is allowed even for non-authenticated users, see https://github.com/solidusio/solidus/blob/aa3063ef7ae577ba0d1b1fdcb5d287445da353e7/api/spec/requests/spree/api/orders_spec.rb#L38. 

If that didn't work, it was maybe because of `Spree::Api::Config.requires_authentication`, which forces authentication on all endpoints?

**Reply by forkata** [reply]
Hey @kennyadsl, thank you for the response here! I did a quick test and I believe you are correct that the if that configuration is set there is no way to do this. For reference the [`authenticate_user`](https://github.com/solidusio/solidus/blob/495772b91f13d8c928041fbcc4fa3e6a54408caa/api/app/controllers/spree/api/base_controller.rb#L51-L59) filter in the API base controller will prevent the action from executing if `Spree::Api::Config.requires_authentication` is set to `true`.
